File: src/main/java/io/github/dqualizer/dqlang/draft/rqaDefinition/loadtest/stimulus/Stimulus.java
package io.github.dqualizer.dqlang.draft.rqaDefinition.loadtest.stimulus;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.draft.rqaDefinition.enums.LoadProfile;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public abstract class Stimulus {
    @JsonProperty("load_profile")
    private LoadProfile loadProfile;

    private int accuracy;

}

File: src/main/kotlin/io/github/dqualizer/dqlang/archive/loadtesttranslator/dqlang/modeling/Stimulus.kt
package io.github.dqualizer.dqlang.archive.loadtesttranslator.dqlang.modeling

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Getter
import lombok.ToString

@Getter
@ToString
class Stimulus {
    @JsonProperty("load_profile")
    private val loadProfile: String? = null

    @JsonProperty("highest_load")
    private val highestLoad: String? = null

    @JsonProperty("time_to_highest_load")
    private val timeToHighestLoad: String? = null

    @JsonProperty("type_of_increase")
    private val typeOfIncrease: String? = null

    @JsonProperty("base_load")
    private val baseLoad: String? = null
    private val accuracy = 0
}
